package_nso_to_oc/xe/xe_system.py

```diff
@@ -1405,6 +1410,86 @@ def cleanup_name_server(config_leftover, name_server_list, vrf_list):
+        "openconfig-system-ext:nat"]
+    oc_system_inside_source = openconfig_system["openconfig-system:system"]["openconfig-system-ext:services"][
+        "openconfig-system-ext:nat"]["openconfig-system-ext:inside"]["openconfig-system-ext:source"]
+    nat_pool = config_before.get("tailf-ned-cisco-ios:ip", {}).get("nat", {}).get("pool")
+    nat_inside_source = config_before.get("tailf-ned-cisco-ios:ip", {}).get("nat", {}).get("inside", {}).get("source")
+    
+    if nat_pool:
+        temp_nat_pool_list = {"openconfig-system-ext:pools": set_nat_pool(nat_pool, config_leftover)}
+        oc_system_nat.update(temp_nat_pool_list)
+
+    if nat_inside_source:
+        temp_nat_inside_source_list = {"openconfig-system-ext:local-addresses-access-lists": set_nat_inside(nat_inside_source, config_leftover)}
+        oc_system_inside_source.update(temp_nat_inside_source_list)
+
+def set_nat_pool(nat_pool, config_leftover):
+    pool_config = {"openconfig-system-ext:pool": []}
+    pool_config_list = pool_config["openconfig-system-ext:pool"]
+    # LIST OF NAT POOLS
+    for index, pool in enumerate(nat_pool):
+        start_address = end_address = "0.0.0.0" # Initialize variables
+        if pool.get("start-address"):
+            start_address = pool["start-address"]
+        if pool.get("end-address"):
+            end_address = pool["end-address"]
+        if pool.get("prefix-length"):
+            temp_pool = {"openconfig-system-ext:name": f'{pool.get("id")}',
+                        "openconfig-system-ext:config": {
+                            "openconfig-system-ext:name": f'{pool.get("id")}',
+                            "openconfig-system-ext:start-address": start_address,
+                            "openconfig-system-ext:end-address": end_address,
+                            "openconfig-system-ext:prefix-length": f'{pool.get("prefix-length")}'
+                        }}
+            pool_config_list.append(temp_pool)
+        elif pool.get("netmask"):
+            temp_pool = {"openconfig-system-ext:name": f'{pool.get("id")}',
+                        "openconfig-system-ext:config": {
+                            "openconfig-system-ext:name": f'{pool.get("id")}',
+                            "openconfig-system-ext:start-address": start_address,
+                            "openconfig-system-ext:end-address": end_address,
+                            "openconfig-system-ext:netmask": f'{pool.get("netmask")}'
+                        }}
+            pool_config_list.append(temp_pool)
+
+    del config_leftover["tailf-ned-cisco-ios:ip"]["nat"]["pool"]
+    return pool_config
+
+def set_nat_inside(nat_inside_source, config_leftover):
+    nat_inside = {"openconfig-system-ext:local-addresses-access-list": []}
+    nat_inside_list = nat_inside["openconfig-system-ext:local-addresses-access-list"]
+    # LIST OF NAT ACLs
+    if nat_inside_source.get("list"):
```

Comment:
This **if** can be skipped with

```suggestion
    for inside_source in nat_inside_source.get("list", []):
```